File: src/NzbDrone.Core/Profiles/AudioScoring/AudioScoreProfile.cs
using System.Collections.Generic;
using NzbDrone.Core.Datastore;

namespace NzbDrone.Core.Profiles.AudioScoring
{
    public class AudioScoreProfile : ModelBase
    {
        public AudioScoreProfile()
        {
            Enabled = true;
            Rules = new List<AudioScoreRule>();
            MutexGroups = new List<AudioScoreMutexGroup>();
        }

        public string Name { get; set; }
        public bool Enabled { get; set; }
        public List<AudioScoreRule> Rules { get; set; }
        public List<AudioScoreMutexGroup> MutexGroups { get; set; }
    }

    public class AudioScoreRule : IEmbeddedDocument
    {
        public string Name { get; set; }
        public string MatchType { get; set; }
        public string Pattern { get; set; }
        public int Score { get; set; }
        public string MutexGroup { get; set; }
        public bool Enabled { get; set; }
    }

    public class AudioScoreMutexGroup : IEmbeddedDocument
    {
        public string Name { get; set; }
        public bool Enabled { get; set; }
    }
}

File: frontend/src/typings/AudioScoreProfile.ts
export interface AudioScoreRule {
  name: string;
  matchType: string;
  pattern: string;
  score: number;
  mutexGroup?: string;
  enabled: boolean;
}

export interface AudioScoreMutexGroup {
  name: string;
  enabled: boolean;
}

interface AudioScoreProfile {
  id: number;
  name: string;
  enabled: boolean;
  rules: AudioScoreRule[];
  mutexGroups: AudioScoreMutexGroup[];
}

export default AudioScoreProfile;
